File: python/get_security_price_technicals_macd.py
from __future__ import print_function
import time
import intrinio_sdk
from intrinio_sdk.rest import ApiException
from pprint import pprint
 
intrinio_sdk.ApiClient().configuration.api_key['api_key'] = 'YOUR_API_KEY'
 
security_api = intrinio_sdk.SecurityApi()

identifier = 'AAPL' # str | A Security identifier (Ticker, FIGI, ISIN, CUSIP, Intrinio ID)

try:
    result = security_api.get_security_price_technicals_macd(identifier)
    indicator = result.indicator
    security = result.security
    technicals = result.technicals

    pprint("Technicals for %s" % security.ticker)
    pprint("%s values for %s returned!" % (len(technicals), indicator.name))
    print("")

    for technical in technicals:
        pprint("DateTime:       %s" % technical.date_time)

        value = technical.value
        pprint("MACD Histogram: %s" % value.macd_histogram)
        pprint("MACD Line:      %s" % value.macd_line)
        pprint("Signal Line:    %s" % value.signal_line)
        pprint("------------------------------------------------------")
except ApiException as e:
    print("Exception when calling SecurityApi->get_security_price_technicals_macd: %s\n" % e)

File: ruby/samples/get_security_price_technicals_macd.rb
# Load the gem
require 'intrinio-sdk'
 
# Setup authorization
Intrinio.configure do |config|
  config.api_key['api_key'] = 'YOUR_API_KEY'
end

security_api = Intrinio::SecurityApi.new

identifier = "AAPL" # String | A Security identifier (Ticker, FIGI, ISIN, CUSIP, Intrinio ID)

opts = {}

begin
  result = security_api.get_security_price_technicals_macd(identifier, opts)
  technicals = result.technicals  
  indicator = result.indicator
  security = result.security

  puts "Technicals for #{security.ticker}"
  puts "#{technicals.size} values for #{indicator.name} returned!"
  puts ""

  technicals.each do |t|
    puts "DateTime:       #{t.date_time}"
    puts "MACD Histogram: #{t.macd_histogram}"
    puts "MACD Line:      #{t.macd_line}"
    puts "Signal Line:    #{t.signal_line}"
    puts "----------------------------------------------------------------"
  end

rescue Intrinio::ApiError => e
  puts "Exception when calling SecurityApi->get_security_price_technicals_macd: #{e}"
end
